src/profile/helpers/editData.ts

```diff
@@ -334,6 +339,51 @@ export function updateItems(
   return dataHasUpdated ? newList : null;
 }
 
+// this is a temporary function to assure primaryItem is always at index 0
+// remove this when backend orders them
+export function movePrimaryAsFirst(allItems: EditData[]): EditData[] {
+  const primaryItemIndex = allItems.findIndex(item => item.primary);
+  if (primaryItemIndex < 1) {
+    return allItems;
+  }
+  const primaryItem = allItems[primaryItemIndex];
+  const firstItem = allItems[0];
+  allItems[0] = primaryItem;
+  allItems[primaryItemIndex] = firstItem;
```

Comment:
Swapping matches quite well what backend returns after save. I know there is no actual sorting done on the backend. Except in emails primary is the first one. Some "groups" (addresses, phones, emails) seem to swap, some just push new primary as first.The discussion below (https://github.com/City-of-Helsinki/open-city-profile-ui/pull/164#discussion_r638519217) will hopefully result in some logic.

I've discussed this with UX-designers and it is fine to swap items even though it does not match save result every time. Re-ordering after save could only occur when user has 3 or more items.

Reply:
Yes, besides the primary being first with emails, there's no sorting in backend. They come in whatever order the database happens to return them. This should be fixed in the backend, so that the order is stable and deterministic.

There's no point trying to mimic the current backend's order in UI, as the order in backend is currently basically random. So whatever might seem to work in some case, fails in others, or the next day.

Reply:
The other conversation about ordering made this irrelevant as code will change.